### ISSUE TITLE
`extended_key_value_attributes` cannot be enabled conditionally

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#82768]
I tried this code:

```rust
#![cfg_attr(
    any(),
    feature(extended_key_value_attributes),
    doc = include_str!("../README.md")
)]
```

I expected to see this happen: The compiler gives no error, because `doc = include_str!` should only be compiled if the predicate is true.

Instead, this happened: The compiler gives an error that `extended_key_value_attributes` is unstable.

I guess this happens because the compiler has to parse the attribute even if the cfg is false? @petrochenkov is giving an error here intentional?

### Meta

`rustc --version`: 1.52.0-nightly (2021-03-01 4f20caa6258d4c74ce6b)

cc https://github.com/rust-lang/rust/issues/78835
_Originally posted by @daxpedda in https://github.com/rust-lang/rust/issues/82539#issuecomment-790418123_

**Comment by petrochenkov** [comment]
`fn process_cfg_attr` parses all "attribute items" inside `cfg_attr` first, and only then checks the predicate and possibly throws the attribute items away. (`extended_key_value_attributes` feature gate is reported during such parsing.)

It's possible to check the predicate first and parse the rest of `cfg_attr` arguments only if the predicate is true, but in that case `cfg_attr` will also accept arbitrary garbage like `#[cfg_attr(FALSE, a b < 7 . / asjd &)]`.

**Comment by petrochenkov** [comment]
The workaround is to add one more layer of `cfg_attr`:
```rust
#![cfg_attr(FALSE, cfg_attr(TRUE, doc = include_str!("../README.md")))] // OK
```

**Comment by jyn514** [comment]
> The workaround is to add one more layer of `cfg_attr`:
> 
> ```rust
> #![cfg_attr(FALSE, cfg_attr(TRUE, doc = include_str!("../README.md")))] // OK
> ```

Thanks! Would it be possible to add that as a structured suggestion in the parser? I can take a look at this weekend if you think it's a good idea :)

**Comment by petrochenkov** [comment]
I don't think it's a reasonable investment of time, especially given that we'll be able to stabilize `extended_key_value_attributes` pretty soon.

**Comment by pksunkara** [comment]
@daxpedda In what scenario are you checking for the feature existence?

In clap, I do:

```rust
#![cfg_attr(feature = "doc", feature(external_doc))]
#![cfg_attr(feature = "doc", doc(include = "../README.md"))]
```

**Comment by jyn514** [comment]
@pksunkara `doc(include)` does not use this feature; see the linked thread: https://github.com/rust-lang/rust/pull/82539#issuecomment-790418123. `doc(include)` is only relevant in that it should be replaced with this feature.

**Comment by pksunkara** [comment]
My bad, I meant to paste the updated one as a probable workaround for people who run into the above bug when they don't need to check the feature existence

```rust
#![cfg_attr(feature = "doc", feature(extended_key_value_attributes))]
#![cfg_attr(feature = "doc", doc = include_str!("../README.md"))]
```

**Comment by jyn514** [comment]
@pksunkara that has the same issue, when the `doc` feature isn't passed it still gives an error that the feature is unstable: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d6f0b2a9d584e889d14fd1688827d510

**Comment by Nemo157** [comment]
So, my understanding of the workaround is that you would write it like this:
```
#![cfg_attr(feature = "doc", feature(extended_key_value_attributes))]
#![cfg_attr(feature = "doc", cfg_attr(feature = "doc", doc = include_str!("../README.md")))]
```

This compiles all the way back to `rustc 1.18.0 (03fc9d622 2017-06-06)`.
